addons/survey/controllers/main.py

```diff
@@ -389,6 +393,9 @@ def _prepare_question_html(self, survey_sudo, answer_sudo, **post):
 
     @http.route('/survey/<string:survey_token>/<string:answer_token>', type='http', auth='public', website=True)
     def survey_display_page(self, survey_token, answer_token, **post):
+        if (lang_code := post.get('lang_code')) and request.httprequest.method == 'POST':
+            return request.redirect(self.env['ir.http']._url_for(f'/survey/{survey_token}/{answer_token}', lang_code))
```

Comment:
I'm not sure I see the benefit of this server-side manipulation...
Why not simply redirect in JS with a `onChangeLanguage` handler on the `select`?

(and let's rename this `**post` into `**kwargs` either way if the route supports GET too, no?)